patternfly/patternfly-elements
# fix(clipboard):

## Pull request description
* Adds a check for window.isSecureContext signifying a secure HTTPS connection ensuring navigator.clipboard API will work correctly. Otherwise, throws an error.
* Hides element using hidden attribute if an error is thrown as the element would be unusable in that state possibly confusing the end-user.
* Hides element using the aria-disabled attribute if the element has been hidden.

## tldr; 
If the site is not secure, and the clipboard fallback cannot be used,  an error is thrown and the element is hidden for all users.

### Related issues

#1868 - [pfe-clipboard] If site isn't HTTPS copy functionality will not work.
#1873 -  bug: pfe-clipboard does not have an accessible experience 


### Preview

1. Check out to the branch fix/CPCORE-6611-pfeClipboard-httpsOnly 
2. Run npm start
3. Go to https://deploy-preview-1981--patternfly-elements.netlify.app/components/clipboard/ or http://localhost:8000/demo/pfe-clipboard on your local

### What has changed and why



- pfe-clipboard.ts
    * add a private state variable, `_secure`, to check for [`window.isSecureContext` value](https://developer.mozilla.org/en-US/docs/Web/API/isSecureContext)
   * add a `_setCopyType` function that sets `copyType` property depending on the value of `_secure`
   * uses `copyType` in `copyTextToClipboard`


### Testing instructions



- In a secure context
  1. Set up the branch locally
  2. Navigate to the pfe-clipboard
  3. User should be able to see and click on the element 
  4. Screen reader should allow you to interact with the element
- In an insecure context
  1. Set up the branch locally
  2. On line 111, set `_secure` to `false`
  3. Navigate to the pfe-clipboard
  4. User should not be able to see or click on the element 
  5. Screen reader should not allow you to interact with the element


#### Browser requirements

Your component should work in all of the following environments:



### Ready-for-merge Checklist





### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**
